x-pack/test/functional/services/machine_learning/api.ts

```diff
@@ -79,5 +141,69 @@ export function MachineLearningAPIProvider({ getService }: FtrProviderContext) {
     async cleanMlIndices() {
       await this.deleteIndices('.ml-*');
     },
+
+    async getJobState(jobId: string): Promise<JobState | undefined> {
+      log.debug(`Fetching job state for job ${jobId}`);
+      const jobStats = await esSupertest
+        .get(`/_ml/anomaly_detectors/${jobId}/_stats`)
+        .expect(200)
+        .then((res: any) => res.body);
+
+      expect(jobStats.jobs).to.have.length(1);
+      const state = jobStats.jobs[0].state as JobState;
+
+      if (state in JobState) {
+        return state;
+      }
+
+      return undefined;
+    },
+
+    async waitForJobState(jobId: string, expectedJobState: JobState) {
+      await retry.waitForWithTimeout(
+        `job state to be ${expectedJobState}`,
+        2 * 60 * 1000,
+        async () => {
+          const state = await this.getJobState(jobId);
+          if (state === expectedJobState) {
+            return true;
+          } else {
+            throw new Error(`expected job state to be ${expectedJobState} but got ${state}`);
+          }
+        }
+      );
+    },
+
+    async getDatafeedState(datafeedId: string): Promise<DatafeedState | undefined> {
+      log.debug(`Fetching datafeed state for datafeed ${datafeedId}`);
+      const datafeedStats = await esSupertest
+        .get(`/_ml/datafeeds/${datafeedId}/_stats`)
+        .expect(200)
+        .then((res: any) => res.body);
+
+      expect(datafeedStats.datafeeds).to.have.length(1);
+      const state = datafeedStats.datafeeds[0].state as DatafeedState;
```

Comment:
same suggestion as before about removing the `as`

Reply:
Done in f32ca31